src/SmartEnum/Core/SmartEnumThen.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Ardalis.SmartEnum.Core
+{
+    public readonly struct SmartEnumThen<TEnum, TValue>
+        where TEnum : SmartEnum<TEnum, TValue>
+        where TValue : IEquatable<TValue>, IComparable<TValue>
+    {
+        private readonly bool isMatch;
+        private readonly SmartEnum<TEnum, TValue> smartEnum;
+        private readonly bool stopEvaluating;
+
+        public SmartEnumThen(bool isMatch, bool stopEvaluating, SmartEnum<TEnum, TValue> smartEnum)
```

Comment:
The constructor can optionally be set to `internal` so that instances can only be created by the methods `When` and `Then`.